miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultIndexService.java

```diff
@@ -10,51 +9,111 @@
+  @Autowired
+  private IndexFamilyService indexFamilyService;
+  @Autowired
+  private AuthorizationManager authorizationManager;
+  @Autowired
+  private DeletionStore deletionStore;
 
   @Override
   public long count(Consumer<String> errorHandler, PaginationFilter... filter) throws IOException {
     return indexStore.count(errorHandler, filter);
   }
 
   @Override
-  public Index getIndexById(long id) {
-    return indexStore.getIndexById(id);
+  public Index get(long id) throws IOException {
+    return indexStore.get(id);
   }
 
   @Override
-  public Collection<IndexFamily> getIndexFamilies() {
-    return indexStore.getIndexFamilies();
+  public List<Index> list(Consumer<String> errorHandler, int offset, int limit, boolean sortDir, String sortCol, PaginationFilter... filter)
+      throws IOException {
+    return indexStore.list(errorHandler, offset, limit, sortDir, sortCol, filter);
+  }
+
+  public void setIndexStore(IndexStore indexStore) {
+    this.indexStore = indexStore;
   }
 
   @Override
-  public Collection<IndexFamily> getIndexFamiliesByPlatform(PlatformType platformType) {
-    return indexStore.getIndexFamiliesByPlatform(platformType);
+  public DeletionStore getDeletionStore() {
+    return deletionStore;
   }
 
   @Override
-  public IndexFamily getIndexFamilyByName(String name) {
-    return indexStore.getIndexFamilyByName(name);
+  public AuthorizationManager getAuthorizationManager() {
+    return authorizationManager;
   }
 
   @Override
-  public List<Index> list(Consumer<String> errorHandler, int offset, int limit, boolean sortDir, String sortCol, PaginationFilter... filter)
-      throws IOException {
-    return indexStore.list(errorHandler, offset, limit, sortDir, sortCol, filter);
+  public SaveDao<Index> getDao() {
+    return indexStore;
   }
 
-  public void setIndexStore(IndexStore indexStore) {
-    this.indexStore = indexStore;
+  @Override
+  protected void loadChildEntities(Index object) throws IOException {
+    loadChildEntity(object.getFamily(), object::setFamily, indexFamilyService);
+  }
+
+  @Override
+  protected void collectValidationErrors(Index object, Index beforeChange, List<ValidationError> errors) throws IOException {
+    if (ValidationUtils.isSetAndChanged(Index::getName, object, beforeChange)
+        && indexStore.getByFamilyPositionAndName(object.getFamily(), object.getPosition(), object.getName()) != null) {
+      errors.add(new ValidationError("name", "This index family already contains an index with this name and position"));
+    }
+    if (ValidationUtils.isSetAndChanged(Index::getPosition, object, beforeChange)
+        && indexStore.getByFamilyPositionAndName(object.getFamily(), object.getPosition(), object.getName()) != null) {
+      errors.add(new ValidationError("position", "This index family already contains an index with this name and position"));
+    }
+    if (beforeChange != null && object.getPosition() != beforeChange.getPosition()) {
```

Comment:
We add duplicate sequences intentionally so that we can have the same index entered multiple times with different box/plate positions in the name :crying_cat_face: